### PR TITLE
external/esp_idf_port: Add ESP32 log module.

### PR DESCRIPTION
**Pull request by ChinaTizenRT**

### CODE REVIEW

File: external/esp_idf_port/log/log.c [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+            s_log_cache[i].level = level;
+            break;
+        }
+    }
+    pthread_mutex_unlock(s_log_mutex);
+}
+
+void clear_log_level_list()
+{
+    while( !SLIST_EMPTY(&s_log_tags)) {
```
Comment (SRCN-LiQin) [inline external/esp_idf_port/log/log.c]: coding rule, blank in () is not consistent here

File: external/esp_idf_port/log/log.c [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+        s_log_default_level = level;
+        clear_log_level_list();
+        pthread_mutex_unlock(s_log_mutex);
+        return;
+    }
+
+    //searching exist tag
+    uncached_tag_entry_t *it = NULL;
+    SLIST_FOREACH( it, &s_log_tags, entries ) {
+        if ( strcmp(it->tag, tag)==0 ) {
```
Comment (SRCN-LiQin) [inline external/esp_idf_port/log/log.c]: coding rule:  add blank around ==

File: external/esp_idf_port/log/log.c [hunk 3]
```diff
@@ -0,0 +1,477 @@
...
+uint32_t esp_log_timestamp() __attribute__((alias("esp_log_early_timestamp")));
+
+#endif //BOOTLOADER_BUILD
+
+void esp_log_buffer_hex_internal(const char *tag, const void *buffer, uint16_t buff_len,
+                        esp_log_level_t log_level)
+{
+    if ( buff_len == 0 ) return;
+    char temp_buffer[BYTES_PER_LINE+3];   //for not-byte-accessible memory
+    char hex_buffer[3*BYTES_PER_LINE+1];
```
Comment (SRCN-LiQin) [inline external/esp_idf_port/log/log.c]: coding rule: add blank around * and + , these 2 lines

File: external/esp_idf_port/log/log.c [hunk 4]
```diff
@@ -0,0 +1,477 @@
...
+
+    do {
+        if ( buff_len > BYTES_PER_LINE ) {
+            bytes_cur_line = BYTES_PER_LINE;
+        } else {
+            bytes_cur_line = buff_len;
+        }
+        if ( !esp_ptr_byte_accessible(buffer) ) {
+            //use memcpy to get around alignment issue
+            memcpy( temp_buffer, buffer, (bytes_cur_line+3)/4*4 );
```
Comment (SRCN-LiQin) [inline external/esp_idf_port/log/log.c]: coding rule: blank around * and +,  also line 385, 397,398, 410,429,433,445

File: external/esp_idf_port/log/log.c [hunk 5]
```diff
@@ -0,0 +1,477 @@
...
+            ptr_line = temp_buffer;
+        } else {
+            ptr_line = buffer;
+        }
+
+        for( int i = 0; i < bytes_cur_line; i ++ ) {
```
Comment (SRCN-LiQin) [inline external/esp_idf_port/log/log.c]: coding rule: no blank for ++ , same issue on other for statement.

**Comment by ChinaTizenRT** [comment]
right, done

**Pull request merged**